pkg/controller/vsphere/reconciler.go

```diff
@@ -442,6 +442,11 @@ func clone(s *machineScope) (string, error) {
 
 	resourcepool, err := s.GetSession().Finder.ResourcePoolOrDefault(s, resourcepoolPath)
 	if err != nil {
+		var multipleFoundError *find.MultipleFoundError
+		if errors.As(err, &multipleFoundError) {
+			return "", machinecontroller.InvalidMachineConfiguration("multiple resource pools found, specify one in config")
+		}
+
```

Comment:
can you describe user stories that make it legit to return the next line as not InvalidMachineConfiguration?

Reply:
I thought about connection errors, when the controller can't reach vSphere endpoints.

Reply:
is there a ways to discriminate that and include it here?